jobrunrpro/src/test/java/org/axonframework/extensions/jobrunrpro/scheduling/JobRunrProEventSchedulerBuilderTest.java

```diff
@@ -0,0 +1,120 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.extensions.jobrunrpro.scheduling;
+
+import org.axonframework.common.AxonConfigurationException;
+import org.axonframework.common.transaction.TransactionManager;
+import org.axonframework.eventhandling.EventBus;
+import org.axonframework.serialization.TestSerializer;
+import org.jobrunr.scheduling.JobScheduler;
+import org.jobrunr.storage.JobSearchRequest;
+import org.jobrunr.storage.Page;
+import org.jobrunr.storage.PageRequest;
+import org.jobrunr.storage.StorageProvider;
+import org.junit.jupiter.api.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class JobRunrProEventSchedulerBuilderTest {
+
+    private static final String TEST_JOB_NAME = "some-job-name";
+    private JobRunrProEventScheduler.Builder builder;
+    private final JobScheduler jobScheduler = mock(JobScheduler.class);
+
+    private final StorageProvider storageProvider = mock(StorageProvider.class);
+    private final TransactionManager transactionManager = mock(TransactionManager.class);
+    private final EventBus eventBus = mock(EventBus.class);
+
+    private final Page<?> page = mock(Page.class);
+
+    @BeforeEach
+    void newBuilder() {
+        reset(page);
+        builder = JobRunrProEventScheduler.proBuilder();
+        doReturn(page).when(storageProvider).getJobs(any(JobSearchRequest.class), any(PageRequest.class));
+    }
+
+    @Test
+    void whenAllPropertiesAreSetCreatesManager() {
+        JobRunrProEventScheduler scheduler = builder
+                .transactionManager(transactionManager)
+                .jobScheduler(jobScheduler)
+                .storageProvider(storageProvider)
+                .eventBus(eventBus)
+                .serializer(TestSerializer.JACKSON.getSerializer())
+                .build();
+
+        assertNotNull(scheduler);
+    }
+
+    @Test
+    void validateNeedsAllPropertiesSet() {
+        builder.jobScheduler(jobScheduler)
+               .transactionManager(transactionManager);
+        assertThrows(AxonConfigurationException.class, () -> builder.build());
+    }
+
+    @Test
+    void whenSettingSchedulerWithNullThrowError() {
+        assertThrows(AxonConfigurationException.class, () -> builder.jobScheduler(null));
+    }
+
+    @Test
+    void whenSettingstorageProviderWithNullThrowError() {
+        assertThrows(AxonConfigurationException.class, () -> builder.storageProvider(null));
+    }
+
+    @Test
+    void whenSettingTransactionManagerWithNullThrowError() {
+        assertThrows(AxonConfigurationException.class, () -> builder.transactionManager(null));
+    }
+
+    @Test
+    void whenSettingEventBusWithNullThrowError() {
+        assertThrows(AxonConfigurationException.class, () -> builder.eventBus(null));
+    }
+
+    @Test
+    void cancelAllInvokesMock() {
```

Comment:
Nit: Being a bit of an ass here, but the `cancelAllInvokesMock` and the `cancelAllWithNameInvokesMock` aren't tests I'd put under the "builder" group. So, I would expect this class to be called `JobRunrProEventSchedulerTest` instead.

Reply:
No longer relevant.